libraries/AC_Fence/AC_Fence.cpp

```diff
@@ -109,15 +110,15 @@ uint8_t AC_Fence::get_enabled_fences() const
 }
 
 // additional checks for the polygon fence:
-bool AC_Fence::pre_arm_check_polygon(const char* &fail_msg) const
+bool AC_Fence::pre_arm_check_polygon(const char* &fail_msg)
 {
     if (!(_enabled_fences & AC_FENCE_TYPE_POLYGON)) {
         // not enabled; all good
         return true;
     }
 
-    if (!_poly_loader.valid()) {
-        fail_msg = "Polygon boundary invalid";
+    if (! _poly_loader.load_from_eeprom()) {
```

Comment:
The only reason is that we don't have a periodic callback into fence to do stuff like loading the fence.

Should I just bite the bullet and actually add that callback into the scheduler for all of the vehicles using AC_Fence?  @rmackay9 ?  I like the idea (primarily because I can leave the prearm checks as `const`), but wasn't sure the vehicle scheduler change would be welcome.

Reply:
sure, i'm totally happy with adding an update() function (or similar) to fences and adding that call into the scheduler.